R/site.R

```diff
@@ -68,3 +68,48 @@ rmd_needs_render <- function(path) {
 rmd_hash <- function(path) {
   digest::digest(path, file = TRUE, algo = "xxhash64")
 }
+
+hugodown_site <- function(input = ".", output_format = NULL) {
+  # read config and output directory
+  config <- hugo_config(input)
+  output_dir <- hugo_config_str(config, "publishdir")
+
+  list(
+    # name for site (used as default name for publishing)
+    name = hugo_config_str(config, "title"),
+
+    # output directory (e.g. "public")
+    output_dir = output_dir,
+
+    # render_site callback
+    render = function(input_file, output_format, envir, quiet, ...) {
+
+      # Render the entire site
+      if (is.null(input_file)) {
```

Comment:
Do you think this should call `rmarkdown::render()` on everything in `site_outdated()`? It seems like that would be a nice gesture to rebuild everything.

Reply:
Yes it should! I just added that behavior.

Reply:
Perfect, thanks!